src/components/ValidateConfigs.ts

```diff
@@ -23,23 +22,17 @@ export class ValidateConfigs extends Component<ValidateConfigProps, {}> {
     }
 
     static validate(props: ValidateConfigProps): string {
-        if (!props.queryNode) {
-            return `${widgetName}: unable to find grid with the name "${props.targetGridName}"`;
-        }
         if (props.inWebModeler) {
             return "";
         }
-        if (!(props.targetGrid && props.targetGrid.declaredClass === "mxui.widget.ListView")) {
-            return `${widgetName}: supplied target name "${props.targetGridName}" is not of the type list view`;
-        }
-        if (!ValidateConfigs.isCompatible(props.targetGrid)) {
+        if (!ValidateConfigs.isCompatible(props.targetListView)) {
             return `${widgetName}: this Mendix version is incompatible with the offline search widget`;
         }
-        if (!props.searchEntity && !ValidateConfigs.isValidAttribute(props.targetGrid._datasource._entity, props)) {
+        if (!props.searchEntity && !ValidateConfigs.isValidAttribute(props.targetListView._datasource._entity, props)) {
             return `${widgetName}: supplied attribute name "${props.searchAttribute}" does not belong to list view`;
         }
         if (props.searchEntity && !ValidateConfigs.itContains(props.searchEntity, "/")) {
-            if (props.searchEntity !== props.targetGrid._datasource._entity) {
+            if (props.searchEntity !== props.targetListView._datasource._entity) {
                 return `${widgetName}: supplied entity "${props.searchEntity}" does not belong to list view data source`;
             }
```

Comment:
i believe the entire validate function should something like this.

```
    static validate(props: ValidateConfigProps): string {
        if (props.inWebModeler) {
            return "";
        }

        if (!ValidateConfigs.isCompatible(props.targetListView)) {
            return `Custom widget ${friendlyId} error : this Mendix version is incompatible with the offline search widget`;
        }

        if (props.searchEntity && !ValidateConfigs.itContains(props.searchEntity, "/")) {
            if (props.searchEntity !== props.targetListView._datasource._entity) {
                return `Custom widget ${friendlyId} error : supplied entity "${props.searchEntity}" does not belong to list view data source`;
            }
        }

        return "";
    }
```

Reply:
Done

Reply:
Changed